embed/etcd.go

```diff
@@ -318,16 +318,25 @@ func (e *Etcd) serve() (err error) {
 		}(l)
 	}
 
-	// Start a client server goroutine for each listen address
-	ch := http.Handler(&cors.CORSHandler{
-		Handler: v2http.NewClientHandler(e.Server, e.Server.Cfg.ReqTimeout()),
-		Info:    e.cfg.CorsInfo,
-	})
+	v2h := func() http.Handler {
```

Comment:
This still needs to be nil'd out if v2 is disabled, otherwise there's still additional overhead in the grpc handler:
```go
var v2h http.Handler
if e.Config().EnableV2 {
    v2h = http.Handler(&cors.CORSHandler{
        Handler: v2http.NewClientHandler(e.Server, e.Server.Cfg.ReqTimeout()),
        Info:    e.cfg.CorsInfo,
    })
}